src/test/java/io/stargate/sgv2/jsonapi/service/cql/builder/QueryBuilderTest.java

```diff
@@ -0,0 +1,177 @@
+package io.stargate.sgv2.jsonapi.service.cql.builder;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import com.bpodgursky.jbool_expressions.Expression;
+import com.bpodgursky.jbool_expressions.Variable;
+import com.datastax.oss.driver.api.core.data.CqlVector;
+import io.stargate.sgv2.jsonapi.service.cql.ExpressionUtils;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.CollectionSettings;
+import io.stargate.sgv2.jsonapi.service.cqldriver.serializer.CQLBindValues;
+import io.stargate.sgv2.jsonapi.service.operation.model.impl.JsonTerm;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class QueryBuilderTest {
+
+  public static final String VECTOR_COLUMN = "query_vector_value";
+  public static final float[] TEST_VECTOR = new float[] {0.1f, 0.2f, 0.3f};
+
+  public static final CqlVector<Float> TEST_CQL_VECTOR = CQLBindValues.getVectorValue(TEST_VECTOR);
+
+  @ParameterizedTest
+  @MethodSource("sampleQueries")
+  @DisplayName("Should generate expected CQL string")
+  public void generateCql(String actualCql, String expectedCql) {
+    assertThat(actualCql).isEqualTo(expectedCql);
+  }
+
+  @SuppressWarnings("PMD.ExcessiveMethodLength")
+  public static Arguments[] sampleQueries() {
```

Comment:
I may have missed this, but did not notice cases where quoting is needed for identifiers... when column name has mixed-case, for example? (quotes around identifier needed to preserve correct identifier casing)
Maybe add one or two cases?

Reply:
added!